crates/ruff_linter/src/rule_selector.rs

```diff
@@ -189,8 +189,9 @@ impl RuleSelector {
     /// Return all matching rules, regardless of rule group filters like preview and deprecated.
     pub fn all_rules(&self) -> impl Iterator<Item = Rule> + '_ {
         match self {
-            RuleSelector::All => RuleSelectorIter::All(Rule::iter()),
-
+            RuleSelector::All => {
+                RuleSelectorIter::All(Rule::iter().filter(|rule| !rule.is_deprecated()))
+            }
```

Comment:
Per the comment prefacing this function, we should not do this filtering here.

Reply:
Hey @WindowGenerator just want to check in again as we get closer to 0.4.0, did you want to work on this still?

Reply:
Hi @zanieb!
Yes, I will do it before the end of the weekend.

Reply:
Not a rush! Thanks for the reply.